meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8mp-SiP-call-for-secondary-boot.patch

```diff
@@ -0,0 +1,54 @@
+From b86e3cf3748729533b0272f514a03f2372e1723f Mon Sep 17 00:00:00 2001
+From: Vanessa Maegima <vanessa.maegima@foundries.io>
+Date: Fri, 18 Mar 2022 14:52:22 -0300
+Subject: [PATCH] plat: imx8mp: SiP call for secondary boot
+
+Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>
+---
+ plat/imx/imx8m/gpc_common.c                  | 13 +++++++++++++
+ plat/imx/imx8m/imx8mp/include/platform_def.h |  2 ++
+ 2 files changed, 15 insertions(+)
+
+diff --git a/plat/imx/imx8m/gpc_common.c b/plat/imx/imx8m/gpc_common.c
+index 6ef4387e2..be1a650c8 100644
+--- a/plat/imx/imx8m/gpc_common.c
++++ b/plat/imx/imx8m/gpc_common.c
+@@ -414,6 +414,19 @@ int imx_src_handler(uint32_t smc_fid, u_register_t x1, u_register_t x2,
+ 		SMC_SET_GP(handle, CTX_GPREG_X1, ret1);
+ 		SMC_SET_GP(handle, CTX_GPREG_X2, ret2);
+ 		break;
++#if defined(PLAT_imx8mp)
++	case IMX_SIP_SRC_SET_SECONDARY_BOOT:
++		if (x2)
++			mmio_setbits_32(IMX_SRC_BASE + SRC_GPR10_OFFSET,
++					SRC_GPR10_PERSIST_SECONDARY_BOOT);
++		else
++			mmio_clrbits_32(IMX_SRC_BASE + SRC_GPR10_OFFSET,
++					SRC_GPR10_PERSIST_SECONDARY_BOOT);
++		break;
++	case IMX_SIP_SRC_GET_SECONDARY_BOOT:
++		val = mmio_read_32(IMX_SRC_BASE + SRC_GPR10_OFFSET);
++		return !!(val & SRC_GPR10_PERSIST_SECONDARY_BOOT);
```

Comment:
is the "!!" necessary?

Reply:
We're just forcing the answer to be '1' or '0'
We are doing the same for all the patches for this support: https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0004-plat-imx8mq-SiP-call-for-secondary-boot.patch#L30

Reply:
ok, Thanks !